pkg/kapp/resources/resources.go

```diff
@@ -401,6 +405,16 @@ func (c *ResourcesImpl) Exists(resource Resource) (bool, error) {
 			return isDone, c.resourceErr(err, "Checking existence of", resource)
 		}
 
+		// Check if we have to compare the UID's also to confirm if it is same resource.
+		if existsOpts.SameUID {
+			if fetchedRes != nil {
```

Comment:
Should we return in case `fetchedRes` is nil to avoid nested if statements?
Although it does feel more readable this way.

Reply:
> Although it does feel more readable this way.

i would stick with current setup as well.

Reply:
> Although it does feel more readable this way.

i would agree current form is more readable.